charts/s1-agent/templates/_helpers.tpl

```diff
@@ -192,7 +192,11 @@ Create the name of the service account to use
 {{- end -}}
 
 {{- define "helper.secret.create" -}}
-{{- empty .Values.secrets.helper_certificate | ternary "true" "" }}
+{{- if or (empty .Values.secrets.helper_certificate) (eq .Values.configuration.deployment_type "argocd") -}}
```

Comment:
I think the following possible as well
{{- or (empty .Values.secrets.helper_certificate) (eq .Values.configuration.deployment_type "argocd") | ternary "true" "" }}